controller/controller.go

```diff
@@ -112,6 +113,38 @@ type Controller struct {
 	Interval time.Duration
 	// The DomainFilter defines which DNS records to keep or exclude
 	DomainFilter endpoint.DomainFilter
+	// Used by RunOnceThrottled to ensure throttling, once per Interval
+	running      bool
+	minStartTime time.Time
+}
+
+var nowValue = struct{}{}
+
+// Make sure execution happens at most once per interval
+func (c *Controller) RunOnceThrottled(ctx context.Context) error {
+	now := time.Now()
+	testing := false
```

Comment:
I would rename this variable. I think that `testing` calls for something that is related to `tests` and this is the code to be tested instead.

Reply:
It is related to tests :) This variable indicates that testing environment is detected so functions returns an error instead of nil

Reply:
I've renamed it to testEnv